### ISSUE TITLE
Skeleton matching network (GAT) doesn't learn anything on custom dataset.

### ISSUE THREAD

**Issue by mateuszk098** [gnns4hri/3D_multi_pose_estimator#1]
Hello @ljmanso, @vangiel, and @pilarbachiller,

First of all, thank you for your work! I'm trying to test this approach on the [AIST](https://google.github.io/aistplusplus_dataset/download.html) dataset, but I've encountered some issues while using `train_skeleton_matching.py`.

Here’s the situation: I prepared a custom dataset by parsing a subset of AIST, following a similar structure to what you used with ARP LAB. Since AIST provides several calibrated cameras, I selected six of them, mirroring your setup with ARP LAB. I then created a pickle file containing a `TransformManager` instance to handle the necessary transformations and added a new configuration to `parameters.py`. After that, I ran `train_skeleton_matching.py`, and the training process started successfully (so far, so good).

However, the network doesn’t seem to learn anything. The loss function remains constant across epochs, and training halts due to early stopping. This behavior seems odd since the network's output should evolve over time, even if we would put there something that wouldn't make sense as an input. I have gone through the learning loop and noticed that `outputs = torch.squeeze(model(feats.float(), subgraph))` variable is always a vector of ones, so this explains why loss doesn't change.

I cross-verified this using ARP LAB, and everything worked as expected - the loss decreased with each epoch. Could you help me identify what might be going wrong? To provide more context, I’ve attached an example of `.json` files along with the configuration I’m using. Each `.json` includes a list of dictionaries with one person seen from six used cameras.

Thank you for your help!

```
parameters = TrackerParameters(
        image_width=1920,
        image_height=1080,
        cameras=[0, 1, 2, 3, 4, 5],
        camera_names=['c01', 'c02', 'c04', 'c05', 'c06', 'c08'],
        fx=[1483.9361119758073, 1561.166571954539, 1604.3931815081523, 1386.5776557502095, 1667.3561978839223, 1565.408341712705],
        fy=[1483.9361119758073, 1561.166571954539, 1604.3931815081523, 1386.5776557502095, 1667.3561978839223, 1565.408341712705],
        cx=[960.0, 960.0, 960.0, 960.0, 960.0, 960.0],
        cy=[540.0, 540.0, 540.0, 540.0, 540.0, 540.0],
        kd0=[0.1373395977080808, 0.16156897795231054, -0.09945789992662554, -0.15615748075841868, -0.052786711044341406, 0.2414519945785073],
        kd1=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
        kd2=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
        p1=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
        p2=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
        joint_list=JOINT_LIST,
        numbers_per_joint=14, 
        numbers_per_joint_for_loss=4,
        transformations_path='../tm_aist_c01c02c04c05c06c08.pickle',
        used_cameras=['c01', 'c02', 'c04', 'c05', 'c06', 'c08'],
        used_cameras_skeleton_matching=['c01', 'c02', 'c04', 'c05', 'c06', 'c08'],
        used_joints = [0, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17],
        min_number_of_views = 2,
        format=FORMAT,
        graph_alternative='3',
        axes_3D = {'X': (0, 1.), 'Y': (1, 1.), 'Z': (2, -1.)} #For drawing the skeletons: each tuple represents (coordinate index, axis direction)
    )
```
[gLH_sBM_cAll_d16_mLH0_ch01.json](https://github.com/user-attachments/files/18006140/gLH_sBM_cAll_d16_mLH0_ch01.json)

**Comment by pilarbachiller** [comment]
Thank you very much for your message. The JSON file you sent looks correct. Could you share the JSON files you are using for training (at least 3 or 4) and the pickle file with the transformations? That way, I can inspect the graphs that are being generated for training to try to find the issue. 

Thank you!

**Comment by pilarbachiller** [comment]
Hello! Yes, we are using meters for the translation vectors of the reference frames. We will clarify this in the documentation. Thank you for pointing it out.

Regarding your questions:

1.   Each keypoint is represented by five elements. The first is the keypoint ID, the second and third are the x and y coordinates, the fourth is a boolean value indicating whether the keypoint is visible or not, and the last is a value between 0 and 1, representing the certainty of the detection for that keypoint. Some skeleton detectors provide this certainty value, which is why we include the field. The fourth value adds flexibility to dataset generation, allowing users to decide whether to include all keypoints (indicating visibility through this field) or only the visible ones.

2. The Timestamp field is used during dataset acquisition to synchronize the different views. The other field refers to the source of the data. In the ARP Lab dataset, the data were acquired directly from the live video flow provided by the cameras, which is why the “live” prefix was used.

3.  The variable “freqs” was included for debugging purposes. It stores the number of graphs for different numbers of people. Its contents are not used but are simply printed. As you correctly pointed out, I have now removed it to avoid confusion and potential errors.

4.  The number of files corresponds to the maximum number of people the network will use during training. Theoretically, the model’s predictions for more people won’t be as accurate as for that maximum number, though we haven’t tested this yet. For optimal results, we recommend training with a number of files that corresponds to the maximum number of people expected in the environment. That said, the same file can be used multiple times, as its contents are shuffled during graph generation. Therefore, the recordings used to train the model do not need to be excessively numerous.

I hope this addresses all your questions. Thank you very much for your comments!

**Comment by mateuszk098** [comment]
Thank you so much for your help!
